core/src/test/java/org/testcontainers/images/OverrideImagePullPolicyTest.java

```diff
@@ -0,0 +1,56 @@
+package org.testcontainers.images;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.testcontainers.DockerRegistryContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.FakeImagePullPolicy;
+import org.testcontainers.utility.MockTestcontainersConfigurationRule;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OverrideImagePullPolicyTest {
+
+    @Rule
+    public MockTestcontainersConfigurationRule config = new MockTestcontainersConfigurationRule();
+
+    private ImagePullPolicy originalInstance;
+
+    private ImagePullPolicy originalDefaultImplementation;
+
+    @Before
+    public void setUp() {
+        this.originalInstance = PullPolicy.instance;
+        this.originalDefaultImplementation = PullPolicy.defaultImplementation;
+        PullPolicy.instance = null;
+        PullPolicy.defaultImplementation = Mockito.mock(ImagePullPolicy.class);
+    }
+
+    @After
+    public void tearDown() {
+        PullPolicy.instance = originalInstance;
+        PullPolicy.defaultImplementation = originalDefaultImplementation;
+    }
+
+    @Test
+    public void simpleConfigurationTest() {
+        Mockito
+            .doReturn(FakeImagePullPolicy.class.getCanonicalName())
+            .when(TestcontainersConfiguration.getInstance())
+            .getImagePullPolicy();
+
+        try (DockerRegistryContainer registry = new DockerRegistryContainer()) {
+            registry.start();
+            GenericContainer<?> container = new GenericContainer<>(registry.createImage()).withExposedPorts(8080);
+            container.start();
+            assertThat(container.getImage())
+                .asString()
+                .contains("imagePullPolicy=org.testcontainers.utility.FakeImagePullPolicy");
```

Comment:
I don't really like the assert of this test. We are asserting against an unrelated implementation detail here (`toString()` of `RemoteDockerImage`).

Two ideas how to improve the test:
* Either make `imagePullPolicy` visible for testing and assert against the field.
* Try to achieve a test similar to `AgeBasedPullPolicyTest` or `ImagePullPolicyTest`, where we check that the expected pull policy is used, by asserting that an image is pulled.

WDYT?

Reply:
> Either make imagePullPolicy visible for testing and assert against the field.

Given we are testing the right `imagePullPolicy` has been applied, this sounds like the one to do.